### PR TITLE
fix: 13335 Fixed incorrect classId in `OnDiskKeySerializer` and `OnDiskValueSerializer` classes

### LINKED ISSUE #13335: OnDiskKeySerializer uses wrong classId for OnDiskKey
### Description

`OnDiskKeySerializer.deserialize()` creates a new instance of `OnDiskKey` using its own `classId` rather than that of `OnDiskKey`.
A typical exception thrown during reconnect or loading a saved state:
```
java.lang.ClassCastException: class com.swirlds.platform.state.merkle.disk.OnDiskKeySerializer cannot be cast to class com.swirlds.virtualmap.VirtualKey (com.swirlds.platform.state.merkle.disk.OnDiskKeySerializer and com.swirlds.virtualmap.VirtualKey are in unnamed module of loader 'app')
``` 

### Steps to reproduce

Run a reconnect test with a Hedera services state. Restart a node with a saved state in a Hedera network.

### Additional context

Regression caused by #12570 

### Hedera network

other

### Version

develop

### Operating system

Linux

### PR DESCRIPTION
**Pull request by imalygin**
**Description**:

This PR fixes incorrect use of `classId`-s in  `OnDiskKeySerializer` and  `OnDiskValueSerializer`. Unfortunately, existing unit tests didn't catch that so I updated the test as well

**Related issue(s)**:

Fixes #13335

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/state/merkle/disk/OnDiskKeySerializerTest.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.state.merkle.disk;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class OnDiskKeySerializerTest {}
```
Comment (OlegMazurov) [inline platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/state/merkle/disk/OnDiskKeySerializerTest.java:21]: What does it test?

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/SerializationTest.java [hunk 2]
```diff
@@ -120,26 +131,98 @@ public void migrate(@NonNull final MigrationContext ctx) {
...
+            serializedBytes = writeTree(originalTree, dir);
+        }
+
+        final MerkleHederaState loadedTree = loadeMerkleTree(schemaV1, serializedBytes);
+
+        assertTree(loadedTree);
+    }
+
+    /**
+     * This test scenario is trickier, and it's designed to reproduce <a href="https://github.com/hashgraph/hedera-services/issues/13335">this issue</a>
```
Comment (OlegMazurov) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/SerializationTest.java:187]:
```suggestion
     * This test scenario is trickier, and it's designed to reproduce <a href="https://github.com/hashgraph/hedera-services/issues/13335">#13335: OnDiskKeySerializer uses wrong classId for OnDiskKey.</a>
```

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/SerializationTest.java [hunk 3]
```diff
@@ -160,6 +243,34 @@ void simpleReadAndWrite() throws IOException, ConstructableRegistryException {
...
+        assertThat(animalState.getForModify(B_KEY)).isEqualTo(BEAR);
+        assertThat(animalState.getForModify(C_KEY)).isEqualTo(CUTTLEFISH);
+        assertThat(animalState.getForModify(D_KEY)).isEqualTo(DOG);
+        assertThat(animalState.getForModify(E_KEY)).isEqualTo(EMU);
+        assertThat(animalState.getForModify(F_KEY)).isEqualTo(FOX);
+        assertThat(animalState.getForModify(G_KEY)).isEqualTo(GOOSE);
+    }
+
+    private static void assertTree(MerkleHederaState loadedTree) {
         // Then, we should be able to see all our original states again
```
Comment (OlegMazurov) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/SerializationTest.java:274]: This comment is stale.

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/SerializationTest.java [hunk 4]
```diff
@@ -160,6 +243,34 @@ void simpleReadAndWrite() throws IOException, ConstructableRegistryException {
                 mock(WritableEntityIdStore.class));
         loadedTree.migrate(1);
 
+        return loadedTree;
+    }
+
+    private MerkleHederaState createMerkleHederaState(Schema schemaV1) {
+        // Given a merkle tree with some fruit and animals and country
```
Comment (OlegMazurov) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/SerializationTest.java:250]: This comment... is not helpful.

**Review by OlegMazurov — DISMISSED** [review]

**Review by cody-littley — DISMISSED** [review]
signing off for hashgraph team (we don't actually own any of this code, so I didn't actually review anything... seek other reviews prior to merging)

File: platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/state/merkle/disk/OnDiskKeySerializerTest.java (hunk 1 above)
Comment (imalygin) [reply platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/state/merkle/disk/OnDiskKeySerializerTest.java:21]: It's empty. I'll remove it

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/SerializationTest.java (hunk 3 above)
Comment (imalygin) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/SerializationTest.java:274]: Agreed. I removed itr

**Review by OlegMazurov — APPROVED** [review]

**Review by cody-littley — APPROVED** [review]

**Pull request merged**